### PR TITLE
Optimize StateVectorLQubitManaged CTOR

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Profiling a simple script similar to
```python
from pennylane_lightning.lightning_qubit_ops import StateVectorC128
import time

for nq in range(2, 32, 2):
    t0 = time.time()
    dt = 0.0
    niter = 0
    while niter < 5 or dt < 1.0:
        niter += 1
        sv = StateVectorC128(nq)
        dt = time.time() - t0
    print(f"{nq}, {dt / niter:1.3e}")
```
it appears some time is wasted in a duplicate `memset` call when invoking the `StateVectorC128` constructor.

**Description of the Change:**
Directly set the 0th element to `1.0`.

**Benefits:**
Faster `StateVectorC128` initialization.
speedup_ctor

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-69660]

### CODE REVIEW

File: pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubitManaged.hpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ class StateVectorLQubitManaged final
         : BaseType{num_qubits, threading, memory_model},
           data_{exp2(num_qubits), ComplexT{0.0, 0.0},
                 getAllocator<ComplexT>(this->memory_model_)} {
-        setBasisState(0U);
+        data_[0U] = {1.0, 0.0};
```
Comment (vincentmr) [inline pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubitManaged.hpp:80]: The memory is already set to 0.0 by the `data_` constructor.

Comment (maliasadi) [inline pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubitManaged.hpp:80]: Amazing! :100:

**Review by maliasadi — APPROVED** [review]
:rocket:

**Review by LuisAlfredoNu — APPROVED** [review]
Nice Job. brief change with good results

**Pull request merged**